test/e2e/framework/framework.go

```diff
@@ -477,6 +477,13 @@ func kubectlExecWithRetry(namespace string, podName, containerName string, args
 				continue
 			}
 		}
+		if err != nil {
```

Comment:
There's already a check for a non-nil error right before this. I would move the strings.Contains check you're adding into that block instead of adding another one, assuming we decide to keep this.